cmd/extract.go

```diff
@@ -0,0 +1,66 @@
+package cmd
+
+import (
+	"fmt"
+	"log"
+	"nlp-tool/internal"
+
+	"github.com/spf13/cobra"
+)
+
+var extractCmd = &cobra.Command{
+	Use:   "extract",
+	Short: "Extract top entities from a given URL",
+	Args:  cobra.MaximumNArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		var url string
+		if len(args) > 0 {
+			url = args[0]
+		} else {
+			var err error
+			url, err = internal.UrlInput()
+			if err != nil {
+				log.Fatalf("Invalid input: %s", err)
+			}
+		}
+
+		validURL, err := internal.ValidateURL(url)
+		if err != nil {
+			log.Fatalf("Invalid URL: %s", err)
```

Comment:
You could use the `os.Exit` to return an exit code to the user instead of return or `log.Fatal`.

Normally, Users of CLI applications uses the exit code to create scripts or automations.